### PR TITLE
Remove unsafe-eval from script-src

### PR DESCRIPTION
**Pull request by ranquild**
Reverts metabase/metabase#33164
Related to https://github.com/metabase/metabase-private/issues/73

Un-reverts the removal of `unsafe-eval` in the `script-src` CSP header. Previously there were 2 issues:
- `ttag` still used `new Function` with plurals sometimes so I had to upgrade the library further
- trend lines were broken

How to verify:
- Open a line chart and add a trend line in the viz settings - it should work
- Go to a collection with line charts - the page should be displayed correctly

Screenshot 2023-08-15 at 12 00 15

### CODE REVIEW

File: frontend/src/metabase/visualizations/lib/trends.js [hunk 1]
```diff
@@ -3,34 +3,35 @@ import moment from "moment-timezone";
...
+  ["*", (...args) => args.reduce((x, y) => x * y)],
+  ["/", (...args) => args.reduce((x, y) => x / y)],
+  ["log", x => Math.log(x)],
+  ["pow", (x, y) => Math.pow(x, y)],
+  ["exp", x => Math.pow(Math.E, x)],
 ]);
 // list of allowed expressions
 const EXPRESSION_IDENTIFIERS = new Set(["x"]);
 
-function compileNode(node) {
```
Comment (ranquild) [inline frontend/src/metabase/visualizations/lib/trends.js:17]: This needs to be factored to avoid calling `new Function`

**Review by tsmacdonald — APPROVED** [review]
+1 for the behavior, the new test, and for not using `eval` (backticks), but I'll leave the rest of the review to frontenders :)

File: frontend/src/metabase/visualizations/lib/trends.js [hunk 2]
```diff
@@ -3,34 +3,35 @@ import moment from "moment-timezone";
...
-    return node;
+    return x;
   }
   throw new Error(`Invalid expression: ${node}`);
 }
 
 export function compileExpression(node) {
-  const compiled = compileNode(node);
-  return new Function("x", `return ${compiled};`);
+  return x => computeExpression(node, x);
```
Comment (kamilmielnik) [inline frontend/src/metabase/visualizations/lib/trends.js:34]: I think we don't need `compileExpression` now, we can just inline it.

There is just one usage in a conditional statement. The `else` block already defines a function inline, so it'll go along nicely.

```diff
--- a/frontend/src/metabase/visualizations/lib/trends.js
+++ b/frontend/src/metabase/visualizations/lib/trends.js
export function getTrendDataPointsFromInsight(insight, xDomain, count = 10) {
 
   let fn;
   if (insight["best-fit"]) {
-    fn = compileExpression(insight["best-fit"]);
+    fn = x => computeExpression(insight["best-fit"], x);
   } else {
     fn = x => x * insight.slope + insight.offset;
   }
```

**Review by kamilmielnik — APPROVED** [review]
Looks good :+1:

File: frontend/src/metabase/visualizations/lib/trends.js (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/visualizations/lib/trends.js]: Agree, fixed

**Review by WiNloSt — APPROVED** [review]
Tested locally. Works as expected. It was a little hard to understand the refactoring at first without trying out before/after code and log return values in both `compileNode` (old) and `computeExpression` (new) functions. After that the change looks logical to me.

**Pull request merged**